### ISSUE TITLE
Build error on OSX

### ISSUE THREAD

**Issue by JustusAdam** [barrucadu/irc-conduit#8]
When building on OSX with stack and lts-8.2 I get the following error:

```
/Users/justusadam/projects/irc-conduit/Network/IRC/Conduit/Lens.hs:37:53: error:
    Invalid type signature: F :: ...
    Should be of form <variable> :: <type>
```

There must be something wrong with the preprocessor I think, because it does not seem to handle your macro properly.

**Comment by barrucadu** [comment]
Ah, that's a pain... I had hit the same issue myself, but had hoped there was just something broken about my set-up. I think I'll just expand the macros myself rather than try to find a work-around.

**Comment by barrucadu** [comment]
Fixed in 7f7d298, and on Hackage as version 0.2.2.1.

**Comment by JustusAdam** [comment]
Well, thank you. I was gonna suggest maybe to try the TemplateHaskell functions from `Control.Lens` to generate the lenses, but this works too.
